src/notebook.c

```diff
@@ -509,6 +537,37 @@ static void show_tab_bar_popup_menu(GdkEventButton *event, GeanyDocument *doc)
 	gtk_container_add(GTK_CONTAINER(menu), menu_item);
 	g_signal_connect(menu_item, "activate", G_CALLBACK(on_close_all1_activate), NULL);
 
+	menu_item = gtk_separator_menu_item_new();
+	gtk_widget_show(menu_item);
+	gtk_container_add(GTK_CONTAINER(menu), menu_item);
+
+		menu_item = ui_image_menu_item_new(GTK_STOCK_PASTE, _("Copy _Basename to Clipboard"));
```

Comment:
The only "issue" (nitpick) is the indentation looks funny here still (on Github), it's fixed in the above new functions since the recent edits.